### PR TITLE
16904 - Add EFT Payment Option

### PR DESCRIPTION
**Pull request by Jxio**
*Issue #:*
https://github.com/bcgov/entity/issues/16904

*Description of changes:*
1. Add EFT Payment Option to Payment methods
2. Add enable_eft flag to store
3. convert PaymentMethod.vue to composition-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/util/config-helper.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ export default class ConfigHelper {
   static paymentsAllowedPerAccountType () {
     return {
       [Account.BASIC]: [ PaymentTypes.CREDIT_CARD, PaymentTypes.ONLINE_BANKING ],
-      [Account.PREMIUM]: [ PaymentTypes.PAD, PaymentTypes.BCOL ],
+      [Account.PREMIUM]: [ PaymentTypes.PAD, PaymentTypes.BCOL, PaymentTypes.EFT ],
```
Comment (Jxio) [inline auth-web/src/util/config-helper.ts:238]: this one need tweaks when api done "the flag the indicates the account is allowed to use EFT."

File: auth-web/src/components/auth/common/PaymentMethods.vue [hunk 2]
```diff
@@ -140,6 +154,14 @@ const PAYMENT_METHODS = {
                   be requested when you are ready to complete a transaction.`,
     isSelected: false
   },
+  [PaymentTypes.EFT]: {
```
Comment (seeker25) [inline auth-web/src/components/auth/common/PaymentMethods.vue:157]: There is already a call into payment_accounts, it should grab the `enable_eft` flag, if that's enabled it should provide this option to the client

File: auth-web/src/util/config-helper.ts (hunk 1 above)
Comment (seeker25) [reply auth-web/src/util/config-helper.ts:238]: There is already a call into payment_accounts somewhere.. , it should grab the enable_eft flag - put it in the store, if that's enabled it should provide this option to the client

File: auth-web/src/components/auth/common/PaymentMethods.vue [hunk 3]
```diff
@@ -82,6 +82,18 @@
                   />
                 </div>
 
+                <!-- EFT -->
+                <div
```
Comment (seeker25) [inline auth-web/src/components/auth/common/PaymentMethods.vue:86]: This needs to be feature flagged as well

Comment (seeker25) [reply auth-web/src/components/auth/common/PaymentMethods.vue:86]: image

File: auth-web/src/components/auth/common/PaymentMethods.vue [hunk 4]
```diff
@@ -193,9 +216,16 @@ const PAYMENT_METHODS = {
...
     LinkedBCOLBanner,
     GLPaymentForm
+  },
+  computed: {
+    ...mapState(useOrgStore, [
+      'currentOrgPaymentDetails'
+    ])
   }
 })
 export default class PaymentMethods extends Vue {
```
Comment (seeker25) [inline auth-web/src/components/auth/common/PaymentMethods.vue:226]: Should probably convert this component into composition-api

**Review by seeker25 — APPROVED** [review]
One small point of converting it into composition-api

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2617-lsa9cply.web.app

**Pull request merged**